### ISSUE TITLE
Update for the Raspberry Pi 500?

### ISSUE THREAD

**Issue by Clafou** [Gadgetoid/pi400kb#50]
Could this be updated to support the new Raspberry Pi 500?
When built and executed on a Pi 500, it prints "Failed to open keyboard device" and some other errors.

**Comment by 57r31** [comment]
built with
```
cmake .. -DKEYBOARD_VID=0x2e8a -DKEYBOARD_PID=0x0010 -DKEYBOARD_DEV=/dev/input/by-id/usb-Raspberry_Pi_Ltd_Pi_500_Keyboard-event-kbd
```
works perfectly on rpi 500

**Comment by Gadgetoid** [comment]
> I wonder if my issue is due to a power problem

It's been a long, long time since I tinkered with any of this, but is it possible you missed:

`Add dtoverlay=dwc2 to /boot/config.txt`

**Comment by Clafou** [comment]
> > I wonder if my issue is due to a power problem
> 
> It's been a long, long time since I tinkered with any of this, but is it possible you missed:
> 
> `Add dtoverlay=dwc2 to /boot/config.txt`

Thank you! I did do that (to be precise, /boot/firmware/config.txt, its new location). I see online of an issue with this on 64-bit systems (https://github.com/raspberrypi/bookworm-feedback/issues/77), perhaps that is the problem.

**Comment by Gadgetoid** [comment]
> Thank you! I did do that (to be precise, /boot/firmware/config.txt, its new location)

Good point, I've only updated this over dozens of python lib installers 😆 I should have known.


Edit: The linked issue looks *very* relevant. Thank you.

I really ought to get myself a Pi 500 so I can actually test this... but no NVMe, c'mon! I'd make a joke about it only being useful as a keyboard, but the keyboard on the Pi 400 is garbage so I don't expect much. (end rant)

**Comment by crystalcommunication** [comment]
> I really ought to get myself a Pi 500 so I can actually test this... but no NVMe, c'mon! I'd make a joke about it only being useful as a keyboard, but the keyboard on the Pi 400 is garbage so I don't expect much. (end rant)

That's the only reason I haven't gotten a Pi 500, myself. The extra RAM and faster CPU are notable advantages over the 400, but the Pi 5 has NVMe and it's about the same price when you factor in the cost of accessories. Every time I've considered getting one, I end up deciding on a regular Pi 5 instead. There is an outline on the board where an M.2 connector could be fitted, I've even heard of someone getting it working by soldering some stuff inside it, but alas, it seems the Raspberry Pi Foundation decided to cut that particular corner before release. Here's to hoping they add it on a hypothetical Pi 600.
